modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1702,264 +1719,549 @@ def tree_reduce(
             reduce_func = self._build_treereduce_func(axis.value, reduce_func)
 
         map_parts = self._partition_mgr_cls.map_partitions(self._partitions, map_func)
-        reduce_parts = self._partition_mgr_cls.map_axis_partitions(
-            axis.value, map_parts, reduce_func
+        reduce_parts = self._partition_mgr_cls.map_partitions_full_axis(
+            map_parts, reduce_func, axis=axis.value
         )
-        return self._compute_tree_reduce_metadata(axis.value, reduce_parts)
+        return self._compute_tree_reduce_metadata(axis.value, reduce_parts, dtypes)
 
-    @lazy_metadata_decorator(apply_axis=None)
-    def map(self, func: Callable, dtypes: Optional[str] = None) -> "PandasDataframe":
+    def _make_init_labels_args(self, partitions, index, columns) -> dict:  # noqa: GL08
+        kw = {}
+        kw["index"], kw["row_lengths"] = (
+            self._compute_axis_labels_and_lengths(0, partitions)
+            if index is None
+            else (index, None)
+        )
+        kw["columns"], kw["column_widths"] = (
+            self._compute_axis_labels_and_lengths(1, partitions)
+            if columns is None
+            else (columns, None)
+        )
+        return kw
+
+    def _prepare_frame_to_broadcast(self, axis, indices):
+        """
+        Compute the indices to broadcast `self` considering `indices`.
+
+        Parameters
+        ----------
+        axis : {0, 1}
+            Axis to broadcast along.
+        indices : dict
+            Dict of indices and internal indices of partitions where `self` must
+            be broadcasted.
+
+        Returns
+        -------
+        dict
+            Dictionary with indices of partitions to broadcast.
+
+        Notes
+        -----
+        New dictionary of indices of `self` partitions represents that
+        you want to broadcast `self` at specified another partition named `other`. For example,
+        Dictionary {key: {key1: [0, 1], key2: [5]}} means, that in `other`[key] you want to
+        broadcast [self[key1], self[key2]] partitions and internal indices for `self` must be [[0, 1], [5]]
+        """
+        sizes = self.row_lengths if axis else self.column_widths
+        return {key: dict(enumerate(sizes)) for key in indices.keys()}
+
+    def map(
+        self,
+        func: Callable,
+        *,
+        axis: Optional[Union[AxisInt, Axis]] = None,
+        other: Optional["PandasDataframe"] = None,
+        full_axis=False,
```

Comment:
why do we need this parameter if we have a separate method for this (`map_full_axis`)?